tests/acceptance/blockchain/interfaces/test_solidity_compiler.py

```diff
@@ -14,56 +14,41 @@
 You should have received a copy of the GNU Affero General Public License
 along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
 """
-from os.path import abspath, dirname
 
-import os
+from pathlib import Path
 
+from nucypher.blockchain.eth.sol.compile.compile import multiversion_compile
+from nucypher.blockchain.eth.sol.compile.constants import DEFAULT_VERSION_STRING
 from nucypher.blockchain.eth.deployers import NucypherTokenDeployer
-from nucypher.blockchain.eth.sol.compile import SolidityCompiler, SourceDirs
-from tests.constants import TEST_CONTRACTS_DIR
-from tests.utils.blockchain import TesterBlockchain
+from nucypher.blockchain.eth.sol.compile.config import ALLOWED_PATHS
 
 
 def test_nucypher_contract_compiled(testerchain, test_registry):
-    # Ensure that solidity smart contacts are available, post-compile.
+    """Ensure that solidity smart contacts are available, post-compile."""
     origin, *everybody_else = testerchain.client.accounts
 
     token_contract_identifier = NucypherTokenDeployer(registry=test_registry, deployer_address=origin).contract_name
     assert token_contract_identifier in testerchain._raw_contract_cache
     token_data = testerchain._raw_contract_cache[token_contract_identifier]
     assert len(token_data) == 1
-    assert "v0.0.0" in token_data
+    assert DEFAULT_VERSION_STRING in token_data
 
 
 def test_multi_source_compilation(testerchain):
-    solidity_compiler = SolidityCompiler(source_dirs=[
-        (SolidityCompiler.default_contract_dir(), None),
-        (SolidityCompiler.default_contract_dir(), {TEST_CONTRACTS_DIR})
-    ])
-    interfaces = solidity_compiler.compile()
-
-    # Remove AST because id in tree node depends on compilation scope
-    for contract_name, contract_data in interfaces.items():
-        for version, data in contract_data.items():
-            data.pop("ast")
+    # TODO: Remove AST because id in tree node depends on compilation scope <<< Still relevant?
+    interfaces = multiversion_compile(solidity_source_dirs=testerchain.SOURCES)
```

Comment:
this does not test initial idea, because for both contract sets used same input tuple and there are no merging between same two equal version of contracts as it was before